lsp/nls/src/linearization/building.rs

```diff
@@ -162,8 +162,41 @@ impl<'b> Building<'b> {
                     .as_option()
                     .and_then(|value_index| self.get_item_kind(current_file, value_index))
             }
+            TermKind::Declaration(_, _, ValueState::Known(value), Some(idents)) => {
+                let item = self.get_item_kind(current_file, *value)?;
+                let item = self.resolve_reference(current_file, item)?;
+
+                let mut ids = idents.clone();
+                // (previous value, current value)
+                let mut result = (item, item);
```

Comment:
IMHO it would be clearer to have two separate variables `prev_value` and `curr_value` (or whatever you call them) rather than using an anonymous tuple. Unless there is a good reason to do so (like calling a stdlib functions that expect a tuple or something).